soroban-sdk/src/testutils/mock_auth.rs

```diff
@@ -0,0 +1,69 @@
+#![cfg(any(test, feature = "testutils"))]
+
+use crate::{contractimpl, xdr, Address, BytesN, RawVal, Vec};
+
+#[doc(hidden)]
+pub struct MockAuthContract;
+
+#[contractimpl(crate_path = "crate")]
+impl MockAuthContract {
+    #[allow(non_snake_case)]
+    pub fn __check_auth(_signature_payload: RawVal, _signatures: RawVal, _auth_context: RawVal) {}
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct MockAuth<'a> {
+    pub address: &'a Address,
+    pub nonce: u64,
```

Comment:
Can we get rid of the nonce here? It's not a relevant detail and it can be easily loaded from the ledger using `read_nonce` host fn. This would not let you use `From` conversions, but I think that's fine.

Reply:
Interesting. It's not obvious to me that it's an irrelevant detail, or always predictable. I'd like to experiment with that idea in a follow up change that tries out a series of scenarios to make sure it's predictable and not relevant, after merging this pull request.

Reply:
It's not relevant because the contract has no impact on it. Adding nonce here is basically covering the host logic, which I really want to avoid for the contract tests. The only valid nonce value currently is the one provided by `read_nonce` utilty.

I also don't want to expose this because in the following release we hopefully should be able to migrate to random temp nonces (https://github.com/stellar/rs-soroban-env/issues/655), which would make this even less relevant than it is now.

If it's really hard to change and if we don't expect much use of this during the release iteration, we probably could leave this for a bit, but I'd really like to avoid exposing this if possible.

Reply:
Issue to track addressing this: https://github.com/stellar/rs-soroban-sdk/issues/935#issue-1696489931